docs/guides/es2k/es2k-guide.md

```diff
@@ -0,0 +1,212 @@
+If you want 16 cfgqs, use "cfgqs-idx": "0-15". Specify the range as a string. Index numbers from 0 to 15 are supported.
+In multi process environment, user should plan and split the queues between primary and secondary processes and specify
+the range with cfgqs-idx parameter. Total number of queues split between processes should not exceed 16.
+
+#### Run the infrap4d daemon
+
+```bash
+cd $IPDK_RECIPE
+sudo ./install/sbin/infrap4d
+```
+
+*Note*: By default, infrap4d runs in detached mode. If you want to run in
+attached mode, specify the --nodetach command-line option.
+
+- All infrap4d logs are by default logged under `/var/log/stratum`.
+- All P4SDE logs are logged in `p4_driver.log` under `$IPDK_RECIPE`.
+*-All OVS logs are logged under `/tmp/ovs-vswitchd.log`.
+
+### Run a sample program
+
+Open a new terminal to set the pipeline and try the sample P4 program.
+Set up the environment and export all environment variables to sudo user.
+
+```bash
+source ./scripts/es2k/setup_env.sh $IPDK_RECIPE $SDE_INSTALL $DEPEND_INSTALL
+./scripts/es2k/copy_config_files.sh $IPDK_RECIPE $SDE_INSTALL
+alias sudo='sudo PATH="$PATH" HOME="$HOME" LD_LIBRARY_PATH="$LD_LIBRARY_PATH" SDE_INSTALL="$SDE_INSTALL"'
+```
+
+#### Create IPDF netdevs
+
+After installing ATE Kernel on HOST machine, install the following drivers
+to bind the network devices (netdevs) to the E2100 target.
+
+```bash
+modprobe auxiliary 
+modprobe idpf
+```
+
+#### Create P4 artifacts
+
+- P4 programs are available in the P4 SDE repository.
+
+- Obtain a copy of the P4 Compiler for ES2K and install it on your
+  development system.
+
+- Set the environment variable OUTPUT_DIR to the location where artifacts
+  should be generated and where p4 files are available
+
+```bash
+export OUTPUT_DIR=/root/p4-driver/mev_reference_p4_files/simple_l3_l4_pna
+```
+
+- Generate the artifacts using the p4c compiler installed in the previous step:
+
+```bash
+p4c-pna-xxp -I/usr/lib -I/usr/share/p4c/p4include -I/usr/share/p4c/idpf-lib \
+            $OUTPUT_DIR/simple_l3_l4_pna.p4 -o $OUTPUT_DIR/simple_l3_l4_pna.s \
+            --p4runtime-files $OUTPUT_DIR/simple_l3_l4_pna.p4info.txt \
+            --context $OUTPUT_DIR/simple_l3_l4_pna.context.json \
+            --bfrt $OUTPUT_DIR/simple_l3_l4_pna.bfrt.json
+```
+
+*Note*: The above commands will generate three files
+(`simple_l3_l4_pna.p4info.txt`, `simple_l3_l4_pna.bfrt.json`, and
+`simple_l3_l4_pna.context.json`).
+
+- Modify `simple_l3_l4_pna.conf` to specify the correct paths for
+  bfrt-config, context, and config.
+
+- Create a dummy tofino.bin file, which is needed by tdi_pipeline_builder.
+
+```bask
+touch tofino.bin
+```
+
+- TDI pipeline builder combines the artifacts generated by the p4c compiler
+  to generate a single bin file to be pushed from the controller.
+  Generate binary executable using the TDI pipeline builder command below:
+
+```bash
+./install/bin/tdi_pipeline_builder \
```

Comment:
every other command has "sudo", should we add sudo to tdi_pipeline_builder as well?